pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -245,45 +248,62 @@ func (gce *GCECloud) targetPoolURL(name, region string) string {
 }
 
 func waitForOp(op *compute.Operation, getOperation func() (*compute.Operation, error)) error {
```

Comment:
Could we just have `waitForOp` not take an operation? Or have `getOperation` be a `func(*compute.Operation) (*compute.Operation, error)`? Then you wouldn't have to check for nil in both places.